src/components/utils/ManagedTree.js

```diff
@@ -46,6 +52,27 @@ let ManagedTree = React.createClass({
     this.setState({ expanded: expanded });
   },
 
+  highlightItem(hLightItems) {
+    const expanded = this.state.expanded;
+    let done = false;
+    let index = hLightItems.length - 1;
+    // Look at folders starting from the top-level until finds a
+    // closed folder and highlight this folder
+    while(!done && index >= 1) {
```

Comment:
perhaps we want to first check to see if the source is visible? if so then we highlight it, if not highlight the lowest directory.